### PR TITLE
[FEATURE] pandas DF conversion tutorial

### PR DESCRIPTION
**Pull request by axelwalter**
Added a section how to use the pandas data frame conversion functionality.

### CODE REVIEW

**Comment by axelwalter** [comment]
Could we do something like this for all our pyOpenMS objects?
https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#auto-document-your-python-code

**Comment by timosachsenberg** [comment]
@rahul799 we had similar discussions a while ago in https://github.com/OpenMS/pyopenms-extra/issues/107
Do you know if it is possible to add this auto document?

File: docs/source/pandas_df_conversion.rst [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+==========================
+
+In pyOpenMS some data structures can be converted to a tabular format as a ``pandas.DataFrame``. This allows convenient access to data and meta values of spectra, features and identifications.
+
+Required imports for the examples:
+
+.. code-block:: python
+
+    import sys
+    sys.path.insert(0, '/home/axel/dev/OpenMS-build/pyOpenMS')
```
Comment (timosachsenberg) [inline docs/source/pandas_df_conversion.rst:11]: change?

File: docs/source/pandas_df_conversion.rst [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+
+.. code-block:: python
+
+    df = feature_map.get_df(meta_values = 'all')
+    df.head(2)
+
+.. csv-table:: feature_map.get_df(meta_values = 'all')
+   :widths: 2 20 5 20 20 20 20 20 20 20 20 20 20 20 20 20 20
+   :header: 	, "sequence",	"charge",	"RT",	"mz",	"RTstart",	"RTend",	"mzstart",	"mzend",	"quality",	"intensity", "FWHM", "spectrum_index", "spectrum_native_id", "label", "score_correlation",	"score_fit"
```
Comment (timosachsenberg) [inline docs/source/pandas_df_conversion.rst:109]: what is sequence here?

File: docs/source/pandas_df_conversion.rst [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+   "1",	"E",	"0",	"1248.33",	"897.449",	"1.2",	"25917900.0",	"25917900.0"
+
+.. code-block:: python
+
+    df = consensus_map.get_intensity_df()
+    df.head(2)
+
+.. csv-table:: consensus_map.get_intensity_df()
+   :widths: 2 30 30
+   :header:     , "data/MapAlignmentFeatureMap1.xml",	"data/MapAlignmentFeatureMap2.xml"
```
Comment (timosachsenberg) [inline docs/source/pandas_df_conversion.rst:187]: the xml extensio is a it strange here? is this featureXML ?

File: docs/source/pandas_df_conversion.rst [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+    df = consensus_map.get_metadata_df()
+    df.head(2)
+
+.. csv-table:: consensus_map.get_metadata_df()
+   :widths: 2 10 20 20 20 20
+   :header:     , "sequence",	"charge",	"RT",	"mz",	"quality"
+            "id"
+
+   "0",	"A",	"0",	"1273.27",	"904.470",	"1.1"
```
Comment (timosachsenberg) [inline docs/source/pandas_df_conversion.rst:203]: maybe a file with more realistic entries would be better here (e.g., not sure but maybe from the ProteomicsLFQ testfiles?)

File: docs/source/pandas_df_conversion.rst [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+        **Parameters:**
+
+        **peps :** 
+        
+        list of PeptideIdentification objects
+
+        **decode_ontology :** default True
+        
+        decode meta value names
```
Comment (timosachsenberg) [inline docs/source/pandas_df_conversion.rst:220]:
```suggestion
        if meta values contain CV identifer (e.g., from PSI-MS) they will be automatically decoded into the human readable CV term name.
```

File: docs/source/pandas_df_conversion.rst [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+        **default_missing_values :** default {bool: False, int: -9999, float: np.nan, str: ''}
```
Comment (timosachsenberg) [inline docs/source/pandas_df_conversion.rst:222]: is there a better way than using -9999 ?

Comment (axelwalter) [reply docs/source/pandas_df_conversion.rst:222]: In numpy np.nan is only for floats, there is no missing value for ints. This is just a fallback, the user can also change.

Comment (timosachsenberg) [reply docs/source/pandas_df_conversion.rst:222]: I see. Ok then let's keep the sentinel

File: docs/source/pandas_df_conversion.rst (hunk 2 above)
Comment (axelwalter) [reply docs/source/pandas_df_conversion.rst:109]: it is the sequence of the best hit peptide for this feature

Comment (timosachsenberg) [reply docs/source/pandas_df_conversion.rst:109]: ok great. 
- maybe also have the getScore() of that hit?
- maybe rename mzstart to MZstart /end

Comment (axelwalter) [reply docs/source/pandas_df_conversion.rst:109]: Hm, then it would be better to update the pyOpenMS code first before releasing this tutorial?

**Comment by rahul799** [comment]
> @rahul799 we had similar discussions a while ago in #107 Do you know if it is possible to add this auto document?

Hi @timosachsenberg 
I will go through it and will try to find some way.

**Comment by axelwalter** [comment]
Added 'BSA1_F1_idmapped.featureXML' as a new example data file, because of that the test code in notebooks will fail.

**Pull request merged**